src/MongoDB/Manager.c

```diff
@@ -81,17 +81,55 @@ PHP_METHOD(Manager, __construct)
 		zval_to_bson(options, PHONGO_BSON_NONE, &bson_options, NULL TSRMLS_CC);
 	}
 
-	uri = php_phongo_make_uri (uri_string, &bson_options TSRMLS_CC);
-	if (uri) {
-		intern->client = php_phongo_make_mongo_client(uri, driverOptions TSRMLS_CC);
-		mongoc_uri_destroy(uri);
-	} else {
-		phongo_throw_exception(PHONGO_ERROR_RUNTIME TSRMLS_CC, "%s", "Failed to parse MongoDB URI");
+	if (!(uri = php_phongo_make_uri(uri_string, &bson_options TSRMLS_CC))) {
+		phongo_throw_exception(PHONGO_ERROR_RUNTIME TSRMLS_CC, "Failed to parse MongoDB URI: '%s'", uri_string);
+		bson_destroy(&bson_options);
+
+		return;
 	}
+
+	intern->client = php_phongo_make_mongo_client(uri, driverOptions TSRMLS_CC);
+	mongoc_uri_destroy(uri);
+
 	if (!intern->client) {
-		phongo_throw_exception(PHONGO_ERROR_RUNTIME TSRMLS_CC, "%s", "Failed to create Manager from URI");
+		char *str;
+		size_t str_len;
+
+		str = bson_as_json(&bson_options, &str_len);
+		phongo_throw_exception(PHONGO_ERROR_RUNTIME TSRMLS_CC, "Failed to create Manager from URI '%s' and options: %s", uri_string, str);
```

Comment:
I reverted to just printing the URI for the two original exceptions (URI parsing fails and client initialization fails). I did change the URI parsing exception to use InvalidArgumentException, which I think makes more sense.

For the later two exceptions, I've changed them to InvalidArgumentException and just report that we couldn't apply the RP or WC options to the Manager.